torch/_refs/nn/functional/__init__.py

```diff
@@ -332,6 +333,27 @@ def tanhshrink(a: TensorLikeType) -> TensorLikeType:
     return refs.sub(a, refs.tanh(a))
 
 
+@register_decomposition(torch.ops.aten.threshold)
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def threshold(
+    a: TensorLikeType, threshold: NumberType, value: NumberType, inplace: bool = False
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.threshold
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    # the straightforward computation drops nan's from input in some cases
+    partial = torch.where(a > threshold, a, value)
```

Comment:
You can do it in one step
```
torch.where(a <= threshold, value, a)
```
which will propagate all nan values from `a`. I'm not certain if it should be `<=` or `<` here, check the grad behavior.

Reply:
oh, is nan propagation asymetric somehow? i did try to do it in one step but with the `>` comparison direction, and nans definitely didn't propagate.

Reply:
Comparison ops when one of the args is `nan` always return `false` so (given that you want only propagate `nans` from `a`) `a` should be on the lhs side of `where`.